File: api/notification/definitions/notification.lua
---@alias NotificationStyle "info" | "success" | "error"

---@class NotificationOptions
---@field style? NotificationStyle
---@field duration? integer

---@class ClientNotificationProvider
---@field showNotification fun(message: string, options?: NotificationOptions): nil

---@class ServerNotificationProvider
---@field showNotification fun(playerId: integer, message: string, options?: NotificationOptions): nil

File: api/notification/definitions/notification.ts
export type NotificationStyle = "info" | "success" | "error";

export interface NotificationOptions {
    style?: NotificationStyle;
    duration?: number;
}

export interface ClientNotificationProvider {
    showNotification(message: string, options?: NotificationOptions): void;
}

export interface ServerNotificationProvider {
    showNotification(playerId: number, message: string, options?: NotificationOptions): void;
}
